component/kafka/group/component.go

```diff
@@ -167,13 +167,19 @@ type Component struct {
 	batchTimeout time.Duration
 	retries      uint
 	retryWait    time.Duration
+	topicChecker func(ctx context.Context, c *Component) error
```

Comment:
This is need to make sure that the `CheckTopic` method is not called before the other `OptionFunc` are called.

Reply:
Why is this a problem?

Reply:
because the `topicChecker` method uses several parameters of the `Component` itself, take this line of the method defined internally to the `CheckTopic` function to see if you see what I mean:

````go
func(ctx context.Context, c *Component) error {
      client, err := sarama.NewClient(c.brokers, c.saramaConfig)
    ...
}
```
by instead setting the `topicChecker` field I am assured to run it after all the fields of `Component` to have been configured according to all `OptionFunc` that are passed to the constructor. 

Makes sense?

````go
func(ctx context.Context, c *Component) error {
      client, err := sarama.NewClient(c.brokers, c.saramaConfig)
    ...
}
```
by instead setting the `topicChecker` field I am assured to run it after all the fields of `Component` to have been configured according to all `OptionFunc` that are passed to the constructor. 

Makes sense?

Reply:
But unless I'm mistaken, the fields we used in `topicCheckers` are passed statically in `group.New`, not via options.

Reply:
Let's break it down. You need:
- Brokers, which is set very early so it is available 
- sarama config, which you do not need to use the component one. You can actually use a new one that fits our needs to query the broker.
With the above it really does not matter when the optionfunc runs.
WDYT?

Reply:
Maybe a bit convoluted but after thinking a bit of pros-cons it felt as the safer approach to me.

Reply:
From the `saramaConfig` I use the `saramaConfig.Metadata.AllowAutoTopicCreation` in the logic which I think should be the same as the component one... Do you see an alternative to that?